data/src/client.rs

```diff
@@ -636,6 +637,16 @@ impl Client {
                             // If we sent (echo) & context exists (we sent from this client), ignore
                             return None;
                         }
+
+                        // send direct message notification
+                        // TODO: check if the active buffer is the same as who sent the message
```

Comment:
@tarkah @casperstorm @andymandias is there existing code/example that checks/references the active/focused buffer's information?

Reply:
Take a look here @hugegulp: https://github.com/squidowl/halloy/blob/dbac8aeb4910095760d93ef19eab109eb3d7864c/src/screen/dashboard/sidebar.rs#L281-L290

`panes` and `focus` is in the `Dashboard` struct. I think what you could do is make a helper `fn` on the `Dashboard` struct which returns the current selected `Pane` (essentially refactoring that piece of code i linked into a `fn`). You can then check the `buffer` of that pane to see if its the one you got a message for.

Once you have such a method you can call it on the `dashboard` which you can get like this:
https://github.com/squidowl/halloy/blob/f0a4e8e4baeea72dfbe1c48f93e29ba1eb03dc70/src/main.rs#L408-L410

Reply:
Alternative, and potentially better, take a look at:

https://github.com/squidowl/halloy/blob/10fde389a92edf4b1f29e2a132abbb771fc88fa0/data/src/history/manager.rs#L250-L265

Reply:
This is a great suggestion. I went with the history manager method! I added the conditional check in `src/main.rs` let me know what you think!